Dockerfile

```diff
@@ -256,18 +282,30 @@ RUN git clone https://github.com/fenics/dolfinx.git && \
     pip3 install . && \
     rm -rf /tmp/*
 
+WORKDIR /root
+
 ########################################
 
 FROM real as notebook
 LABEL description="DOLFIN-X Jupyter Notebook"
 WORKDIR /root
-RUN pip3 install jupyter
-ENTRYPOINT ["jupyter", "notebook", "--ip", "0.0.0.0", "--no-browser", "--allow-root"]
+
+ARG TINI_VERSION=v0.18.0 
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
+RUN chmod +x /tini && \
+    pip3 install jupyter
+
+ENTRYPOINT ["/tini", "--", "jupyter", "notebook", "--ip", "0.0.0.0", "--no-browser", "--allow-root"]
 
 ########################################
 
 FROM complex as notebook-complex
 LABEL description="DOLFIN-X (complex mode) Jupyter Notebook"
+
+ARG TINI_VERSION=v0.18.0 
+ADD https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini /tini
```

Comment:
Do we need this? It would be nice to keep everything as vanilla as possible.

Reply:
Yes, see e.g. all of the official Jupyter notebook stacks:

https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile

Reply:
It is also recommended by Docker themselves:

https://github.com/docker-library/official-images#init